src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/BlobManager.java

```diff
@@ -204,7 +206,32 @@ private void moveFileToRejectedContainer(
             ? AccessCondition.generateLeaseCondition(leaseId)
             : AccessCondition.generateEmptyCondition();
 
-        inputBlob.deleteIfExists(DeleteSnapshotsOption.NONE, deleteCondition, null, null);
+
+        //TODO, for testing, will delete
+        try {
+            TimeUnit.MINUTES.sleep(1);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        try {
+            inputBlob.deleteIfExists(DeleteSnapshotsOption.NONE, deleteCondition, null, null);
+        } catch (StorageException e) {
+            //if lease lost retry
+            log.warn(
+                "Deleting File {} got error, Error code {}, Http status {} ",
+                fileName,
+                e.getErrorCode(),
+                e.getHttpStatusCode(),
+                e
+            );
+
+            if (e.getHttpStatusCode() == HttpURLConnection.HTTP_PRECON_FAILED) {
+                log.info("Deleting File {} got error, retrying...", fileName);
+                inputBlob.deleteIfExists(DeleteSnapshotsOption.NONE, null, null, null);
```

Comment:
why is `deleteCondition` null this time?

Reply:
because we lost the lease , there is no Id to in blobstorage to compare.

Reply:
If the lease is lost an some other thread/process acquires the lease then this will still fail ?

Reply:
we check metadata